crates/ty/src/lib.rs

```diff
@@ -298,48 +298,50 @@ impl MainLoop {
 
                         let mut stdout = stdout().lock();
 
-                        if result.is_empty() {
-                            writeln!(stdout, "{}", "All checks passed!".green().bold())?;
-
-                            if self.watcher.is_none() {
-                                return Ok(ExitStatus::Success);
-                            }
-                        } else {
-                            let mut max_severity = Severity::Info;
-                            let diagnostics_count = result.len();
-
-                            for diagnostic in result {
-                                write!(stdout, "{}", diagnostic.display(db, &display_config))?;
-
-                                max_severity = max_severity.max(diagnostic.severity());
-                            }
-
-                            writeln!(
-                                stdout,
-                                "Found {} diagnostic{}",
-                                diagnostics_count,
-                                if diagnostics_count > 1 { "s" } else { "" }
-                            )?;
-
-                            if max_severity.is_fatal() {
-                                tracing::warn!(
-                                    "A fatal error occurred while checking some files. Not all project files were analyzed. See the diagnostics list above for details."
-                                );
-                            }
-
-                            if self.watcher.is_none() {
-                                return Ok(match max_severity {
-                                    Severity::Info => ExitStatus::Success,
-                                    Severity::Warning => {
-                                        if terminal_settings.error_on_warning {
-                                            ExitStatus::Failure
-                                        } else {
-                                            ExitStatus::Success
+                        if std::env::var("TY_MEMORY_REPORT").as_deref() == Ok("mypy_primer") {
```

Comment:
This is a hack for now, we should have a way to silence diagnostics completely (but keep the memory usage report).